src/Directory.Build.targets

```diff
@@ -26,4 +26,99 @@
+      <_BinariesInTestDirectory Include="$(TargetDir)*.dll" />
+      <_BinariesInTestDirectory Update="@(_BinariesInTestDirectory)" IntermediatePdb="%(RootDir)%(Directory)%(Filename).ppdb" />
+    </ItemGroup>
+
+    <PropertyGroup Condition="'$(_UseOpenCover)' == 'true'">
+      <_PdbConverterPath>$(NuGetPackageRoot)microsoft.diasymreader.pdb2pdb\$(MicrosoftDiaSymReaderPdb2PdbVersion)\tools\Pdb2Pdb.exe</_PdbConverterPath>
+    </PropertyGroup>
+
+    <Exec Command="&quot;$(_PdbConverterPath)&quot; &quot;%(_BinariesInTestDirectory.Identity)&quot; /out &quot;%(_BinariesInTestDirectory.IntermediatePdb)&quot; /extract"
+          Condition="'$(_UseOpenCover)' == 'true'"
+          IgnoreExitCode="true" />
+
+    <!-- Convert portable PDBs to Windows PDBs for OpenCover -->
+    <ItemGroup Condition="'$(_UseOpenCover)' == 'true'">
+      <_PortablePdb Include="$(TargetDir)*.ppdb" />
+      <_PortablePdb Update="@(_PortablePdb)" Binary="%(RootDir)%(Directory)%(Filename).dll" OutputPdb="%(RootDir)%(Directory)%(Filename).pdb" />
+    </ItemGroup>
+
+    <Exec Command="&quot;$(_PdbConverterPath)&quot; &quot;%(_PortablePdb.Binary)&quot; /pdb &quot;%(_PortablePdb.Identity)&quot; /out &quot;%(_PortablePdb.OutputPdb)&quot;"
+          Condition="'$(_UseOpenCover)' == 'true'" />
+
+    <PropertyGroup Condition="'$(_RunOnCore)' != 'true'">
+      <_XUnitConsoleExe>xunit.console.exe</_XUnitConsoleExe>
+      <_XUnitConsoleExe Condition="'$(_TestArchitecture)' == 'x86'">xunit.console.x86.exe</_XUnitConsoleExe>
+      <_XUnitRunnerCommand>$(NuGetPackageRoot)xunit.runner.console\$(XUnitVersion)\tools\net452\$(_XUnitConsoleExe)</_XUnitRunnerCommand>
+      <_XUnitRunnerCommandArgs>"$(TargetPath)" -noshadow -xml "$(_TestResultsXmlPath)" -html "$(_TestResultsHtmlPath)" $(XUnitRunnerAdditionalArguments)</_XUnitRunnerCommandArgs>
+      <_TestRunnerCommand>"$(_XUnitRunnerCommand)" $(_XUnitRunnerCommandArgs)</_TestRunnerCommand>
+      <_TestResultsDisplayPath>$(_TestResultsHtmlPath)</_TestResultsDisplayPath>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(_UseOpenCover)' == 'true'">
+      <_OpenCoverConsoleExe>OpenCover.Console.exe</_OpenCoverConsoleExe>
+      <_OpenCoverCommand>$(NuGetPackageRoot)opencover\$(OpenCoverVersion)\tools\$(_OpenCoverConsoleExe)</_OpenCoverCommand>
+      <_OpenCoverCommandArgs>-register:user -returntargetcode -hideskipped:All -filter:"+[*]*" -excludebyattribute:*.ExcludeFromCodeCoverage* -excludebyfile:*\*Designer.cs -output:"$(_TestResultsOpenCoverPath)"</_OpenCoverCommandArgs>
+      <_TestRunnerCommand>"$(_OpenCoverCommand)" $(_OpenCoverCommandArgs) -target:"$(_XUnitRunnerCommand)" -targetargs:"$(_XUnitRunnerCommandArgs.Replace(`"`, `\"`))"</_TestRunnerCommand>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(ArtifactsTestResultsDir)"/>
+    <Delete Files="$(_TestOutPathNoExt).*" />
+
+    <Message Text="Running tests: $(TargetPath) [$(_TestEnvironment)]" Importance="high"/>
```

Comment:
Why do you redefine the test call instead say just running before and after the Test target?

Reply:
It wraps (replaces) the xunit executable used for running tests. When RepoToolset is updated to support code coverage, this entire target can be removed.